cmd/init-osm-controller/init-osm-controller_test.go

```diff
@@ -59,3 +59,11 @@ func TestCreateDefaultMeshConfig(t *testing.T) {
 	assert.Equal(meshConfig.Spec.Observability.EnableDebugServer, false)
 	assert.Equal(meshConfig.Spec.Certificate.ServiceCertValidityDuration, "24h")
 }
+
+func TestValidateCLIParams(t *testing.T) {
+	assert := tassert.New(t)
+
+	err := validateCLIParams()
```

Comment:
This seems fine as long as other tests in this package are careful not to mess too much with the global state this test is reading.

In general though, I think tests should be setting the global state they need, even if that state is the default. Otherwise it's not at all clear what the test is verifying. And if it's manipulating global state, then it should save the old state and reset it after the test so it doesn't affect other tests. That of course is a lot to consider, so refactoring this in the future to eliminate global state where we can may be worthwhile.

Reply:
Issues created: https://github.com/openservicemesh/osm/issues/3547